src/Text.js

```diff
@@ -1,5 +1,22 @@
-import {withSystemProps, TYPOGRAPHY, COMMON} from './system-props'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import theme from './theme'
+import {TYPOGRAPHY, COMMON, Base} from './constants'
 
-const Text = withSystemProps('span', [...TYPOGRAPHY, ...COMMON])
+const Text = styled(Base)`
+  ${TYPOGRAPHY} ${COMMON};
+`
+
+Text.defaultProps = {
+  is: 'span',
+  theme
+}
+
+Text.propTypes = {
+  ...TYPOGRAPHY.propTypes,
+  ...COMMON.propTypes,
+  is: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
```

Comment:
Should this be `PropTypes.node` instead of `PropTypes.func`? As far as I can tell the way the `is` prop is implemented should mean any element or component can be passed, not just function components.

Reply:
Good catch!